biogeochem/EDCohortDynamicsMod.F90

```diff
@@ -1324,6 +1388,38 @@ subroutine fuse_cohorts(currentSite, currentPatch, bc_in)
                                       currentCohort%frmort = (currentCohort%n*currentCohort%frmort + nextc%n*nextc%frmort)/newn
 
+                                      ! Nutrient fluxes
+                                      currentCohort%daily_n_uptake = (currentCohort%n*currentCohort%daily_n_uptake + & 
+                                           nextc%n*nextc%daily_n_uptake)/newn
+                                      currentCohort%daily_p_uptake = (currentCohort%n*currentCohort%daily_p_uptake + & 
+                                           nextc%n*nextc%daily_p_uptake)/newn
+
+                                      currentCohort%daily_p_demand = (currentCohort%n*currentCohort%daily_p_demand + & 
+                                           nextc%n*nextc%daily_p_demand)/newn
+                                      currentCohort%daily_n_demand = (currentCohort%n*currentCohort%daily_n_demand + & 
+                                           nextc%n*nextc%daily_n_demand)/newn
+                                      
+                                      currentCohort%daily_c_efflux = (currentCohort%n*currentCohort%daily_c_efflux + & 
+                                           nextc%n*nextc%daily_c_efflux)/newn
+                                      currentCohort%daily_n_efflux = (currentCohort%n*currentCohort%daily_n_efflux + & 
+                                           nextc%n*nextc%daily_n_efflux)/newn
+                                      currentCohort%daily_p_efflux = (currentCohort%n*currentCohort%daily_p_efflux + & 
+                                           nextc%n*nextc%daily_p_efflux)/newn
+                                      
+                                      currentCohort%daily_n_need1 = (currentCohort%n*currentCohort%daily_n_need1 + & 
+                                           nextc%n*nextc%daily_n_need1)/newn
+                                      currentCohort%daily_n_need2 = (currentCohort%n*currentCohort%daily_n_need2 + & 
+                                           nextc%n*nextc%daily_n_need2)/newn
+                                      currentCohort%daily_p_need1 = (currentCohort%n*currentCohort%daily_p_need1 + & 
+                                           nextc%n*nextc%daily_p_need1)/newn
+                                      currentCohort%daily_p_need2 = (currentCohort%n*currentCohort%daily_p_need2 + & 
+                                           nextc%n*nextc%daily_p_need2)/newn
+
+                                      ! This carbon variable needs continuity from day to day, as resp_m_def
+                                      ! needs to hold mass and be conservative
+                                      currentCohort%resp_m_def = (currentCohort%n*currentCohort%resp_m_def + & 
+                                           nextc%n*nextc%resp_m_def)/newn
+                                      
```

Comment:
yes, there should be.  The value is not used after cohort fusion for mechanistic purposes, however a history writing call will.  Below I highlight the line where we call the rapid-timestep fates history updates, and it does occur after fusion (it would be better to call this before the dynamics call to fates (and we do that in ELM), but until we fully coupled nutrients with CLM, it would be too messy to swap the order on the CLM side.

https://github.com/ESCOMP/CTSM/compare/fates_main_api...rgknox:fates_cnp_v0?expand=1#diff-8b7656a5ffc6b97053c9b0e70595a2b4R1058

Reply:
I started to add a bunch of the high-frequency carbon balance variables to the cohort merging loop.  I then decided it was better to just re-evaluate when where we call the update to these diagnostics in the history, and make sure that is before we call dynamics.  Therefore, those cohort level diagnostics (everything with "tstep" in the name) are not relevant and don't need to be fused.

The call sequence in E3SM and CTSM's drivers both have this fixed:  https://github.com/ESCOMP/CTSM/pull/1148/files#diff-8b7656a5ffc6b97053c9b0e70595a2b4